src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -560,6 +561,28 @@ public void Close()
             }
         }
 
+        public void SetNonBlocking()
+        {
+            //
+            // Our sockets may start as blocking, and later transition to non-blocking, either because the user
+            // explicitly requested non-blocking mode, or because we need non-blocking mode to support async
+            // operations.  We never transition back to blocking mode, to avoid problems synchronizing that
+            // transition with the async infrastructure.
+            //
+            // Note that there's no synchronization here, so we may set the non-blocking option multiple times
+            // in a race.  This should be fine.
```

Comment:
It looks fine to me as well.  The only thing I notice is that under the covers SetIsNonBlocking is using fcntl, and it has to get the current setting, modify it, and then store it back.  We're not currently using fcntl for any other setting values, but if we were to in the future, this would seem to then be problematic, as at that point multiple threads might try to set different options concurrently, with only one of them actually happening.